drivers/eeprom/eeprom_lpc11u6x.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lpc11u6x_eeprom
```

Comment:
Hi Henrik,
I don't know how much the IAP is shared between all the LPCs. Depending on the MCUs some feature may be available or not. The entry codes and the return status may be different as well. I think @MaureenHelm is the one able to answer this.

Reply:
I just had a quick look at the IAP of the lpc15xx, see https://www.nxp.com/docs/en/user-guide/UM10736.pdf pg. 572. It seems that the command codes, return values etc are the same. Only the iap entry location seems to be different:
```
#define IAP_LOCATION 0x0300 0205
```

Reply:
> I just had a quick look at the IAP of the lpc15xx, see https://www.nxp.com/docs/en/user-guide/UM10736.pdf pg. 572. It seems that the command codes, return values etc are the same. Only the iap entry location seems to be different:
> 
> ```
> #define IAP_LOCATION 0x0300 0205
> ```

Hi @chrta, Thanks for looking.

So should we move it under soc/arm/nxp_lpc/common right now ? Or shall we rather wait for someone adding support for another LPC SoC needing IAP to do that ?

Reply:
@agansari can you take a look at this?

Reply:
> > I just had a quick look at the IAP of the lpc15xx, see https://www.nxp.com/docs/en/user-guide/UM10736.pdf pg. 572. It seems that the command codes, return values etc are the same. Only the iap entry location seems to be different:
> > ```
> > #define IAP_LOCATION 0x0300 0205
> > ```
> 
> Hi @chrta, Thanks for looking.
> 
> So should we move it under soc/arm/nxp_lpc/common right now ? Or shall we rather wait for someone adding support for another LPC SoC needing IAP to do that ?

I have some devboards based on LPC1549 so if you need someone to make some tests and etc I am at your disposal.

Reply:
@mbittan @MaureenHelm the IAP software has the same command structure across older LPC devices; it's used for flash memories most of the time, but in the case if LPC11U6x it's used for eeprom. I see no flash programming in this pull request.
EEPROM driver should be separate from flash, even if they use the same underlying firmware for Zephyr's driver sake, sdk drivers are separate as well.
As bare metal drivers that don't rely on hal_nxp module they are correct and I would approve, but I have not reviewed in detail the other drivers and configurations.
In case flash will be used, a driver that also fixes #6040 and relies on hal/nxp/mcux/drivers/lpc/fsl_flashiap.h would be preferred.

Reply:
@agansari We don't plan on adding support for the flash controller right now, do you think we should keep the EEPROM driver as it is now ?

Reply:
@mbittan yes, there is no other LPC device using eeprom with Zephyr support for now. Also:

> EEPROM driver should be separate from flash, even if they use the same underlying firmware for Zephyr's driver sake, sdk drivers are separate as well.